plugins/modules/nmcli.py

```diff
@@ -1570,6 +1614,16 @@ def __init__(self, module):
 
         self.edit_commands = []
 
+        self.extra_options_validation()
+
+    def extra_options_validation(self):
+        """ Additional validation of options set passed to module that cannot be implemented in module's argspecs. """
+        if self.type not in ("bridge-slave", "team-slave", "bond-slave"):
+            if self.master is not None and self.slave_type is None:
+                self.module.fail_json(msg="'slave_type' option is required when 'master' is specified.")
```

Comment:
The above three lines make the module fail if `master` is specified and `type` is not one of the three mentioned types, and `slave_type` isn't specified. The module before this PR accepted this configuration without failing if I understand it correctly.

Reply:
Fixed.